### PR TITLE
fix(forms/rjsf): add support for modal and drawers

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

Old form docs is still the first documentation in surge.
Old form are broken in drawer.
Apps use custom css to make it look nice (but they don t follow the guideline).

**What is the chosen solution to this problem?**

Move `Form` doc in surge to the last one. Make core stories first. 
Make old form follow the guideline in modal and drawer

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [x] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/forms/stories-core/layouts.js [hunk 1]
```diff
@@ -1,9 +1,30 @@
...
+
 import Form from '../src/index';
 
 const simple = require('./json/concepts/core-simple.json');
+const old = require('../stories/json/simple.json');
+
+const actions = [
+	{
+		name: 'cancel',
+		style: 'link',
```
Comment (annilsson) [inline packages/forms/stories-core/layouts.js]: I think this should be `inverse` & `default`? (it affects alignment if we use link style)

File: packages/forms/stories-core/layouts.js [hunk 2]
```diff
@@ -61,4 +82,26 @@ export default [
...
+		story: () => (
+			<div>
+				<h1>Form (old) by default take 100% with of the container</h1>
+				<Form data={old} />
+			</div>
+		),
+	},
+	{ name: 'drawer-old', story: () => <LayoutDrawer title="Form (old) in a drawer" data={old} /> },
+	{
+		name: 'drawer-stacked-old',
```
Comment (annilsson) [inline packages/forms/stories-core/layouts.js]: Cancel is not aligned with the form fields on the left side

File: packages/forms/stories-core/layouts.js [hunk 3]
```diff
@@ -1,9 +1,30 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { HeaderBar, Layout, Dialog, Drawer } from '@talend/react-components';
+import { action } from '@storybook/addon-actions';
+
 import Form from '../src/index';
 
 const simple = require('./json/concepts/core-simple.json');
+const old = require('../stories/json/simple.json');
```
Comment (annilsson) [inline packages/forms/stories-core/layouts.js]: I can't figure out why (maybe coincidence), but the stories that use `old` data have the buttons left-aligned. Should they not be right-aligned?

Comment (jmfrancois) [reply packages/forms/stories-core/layouts.js]: if you take the OLD stories you can see the submit btn is always left aligned ...

**Comment by jmfrancois** [comment]
rage quit

**Pull request closed without merging**